File: cloud/app/controllers/RestThing.java
package controllers;

import models.Thing;
import play.mvc.Controller;
import play.mvc.Result;

public class RestThing extends Controller {
  
  public static Result register(String url) {
    Thing.register(url);
    return ok("Registering: " + url);
  }
    
  public static Result remove(String id) {
    if(Thing.remove(id)) {
      return ok("Thing removed");
    } else {
      return notFound("Thing not found");
    }  
  }
  
}

File: cloud/app/controllers/RestThing.scala
package controllers

import models.Thing
import play.api._
import play.api.mvc._
import play.api.mvc.Results._
import play.libs.F.Promise

object RestThing extends Controller {
    
  def register(url: String) = Action {
    Ok(Thing.register(url).toString) 
  }
    
  def remove(id: Long) = Action {
   Thing.delete(id)
   Ok("Removed")
  }
	
}
